### PR TITLE
Add DetailViewItemAction

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | yes 
| Fixed tickets | -
| Related issues/PRs | implemented for #6938
| License | MIT
| Documentation PR | sulu/sulu-docs# 

#### What's in this PR?

Add ResourceDetailViewItemAction.

#### Why?

The ResourceDetailViewItemAction allow to deeplink from a list with different recources in it to a detail view.

#### Usage

```php
           ->addItemActions([
                new ListItemAction(
                    'detail_link',
                    [
                        'resource_key_property' => 'referenceResourceKey',
                        'resource_id_property' => 'referenceResourceId',
                        'resource_view_attributes_property' => 'referenceViewAttributes',
                    ],
                ),
```

### CODE REVIEW

**Comment by alexander-schranz** [comment]
@Prokyonn keep an focus on the newly added methods and refactoring not 100% sure about the naming and the added new services and service methods.

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        this.clear();
+    }
+
+    clear() {
+        this.resourceViews = {};
+    }
+
+    addResourceViews(resourceViews: ResourceViewsMap) {
+        Object.keys(resourceViews).forEach((resourceKey) => {
+            if (this.resourceViews[resourceKey]) {
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js:17]: When we already have a `has` method, I would use it here

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return this.resourceViews[resourceKey].views[view] !== undefined;
+    }
+
+    get(view: string, resourceKey: string): string {
+        if (typeof this.resourceViews[resourceKey] !== 'object') {
+            throw new Error('The resource "' + resourceKey + '" was not found.');
+        }
+
+        if (typeof this.resourceViews[resourceKey].views[view] !== 'string') {
+            throw new Error('The resource view "' + view + '" for resource "' + resourceKey + '" was not found.');
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js:39]: Do we really have to distinguish in the error between `'The resource "' + resourceKey + '" was not found.` and `'The resource view "' + view + '" for resource "' + resourceKey + '" was not found.'` ?

I think the second one should be sufficient and then we could here also use the `has` method

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/List/itemActions/ResourceDetailLinkItemAction.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+// @flow
+import AbstractListItemAction from './AbstractListItemAction';
+
+export default class ResourceDetailLinkItemAction extends AbstractListItemAction {
```
Comment (Prokyonn) [inline src/Sulu/Bundle/AdminBundle/Resources/js/views/List/itemActions/ResourceDetailLinkItemAction.js]: Maybe just name it `DetailLinkItemAction` ? Would be a bit shorter 🤔

**Comment by Prokyonn** [comment]
> @Prokyonn keep an focus on the newly added methods and refactoring not 100% sure about the naming and the added new services and service methods.

I think the naming makes sense, with `view` in it, it should be quite clear what routes are registered here

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js:17]: The `has` is specific for the `view` of a resource key not if general the resourceKey exist.

File: src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js (hunk 2 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/registries/resourceViewRegistry.js:39]: Think if the error happens it is good to know if the `resourceKey` was not provided or if the `view` is just missing, think for better DX it make sense to have different specific error messages here.

File: src/Sulu/Bundle/AdminBundle/Resources/js/views/List/itemActions/ResourceDetailLinkItemAction.js (hunk 3 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/List/itemActions/ResourceDetailLinkItemAction.js]: Okay then we name it also `detail_link` in the PHP Code?

Comment (Prokyonn) [reply src/Sulu/Bundle/AdminBundle/Resources/js/views/List/itemActions/ResourceDetailLinkItemAction.js]: yes I think it is a bit shorter and cleaner, and we don't lose important information

**Review by Prokyonn — APPROVED** [review]

**Pull request merged**